### PR TITLE
Not use `transform_variance` for unscaled targets

### PR DESCRIPTION
**Pull request by shihchengli**
## Description
Address #1107

## Checklist
- [ ] linted with flake8?
- [ ] (if appropriate) unit tests added?

### CODE REVIEW

**Comment by jjacobgreen** [comment]
Looks good to me. Have tested against my failing minimal reproducible example and it's fixed the issue.

**Comment by jjacobgreen** [comment]
I don't know how feasible it is, but adding an [extra test case with a predictor that has uncertainty](https://github.com/chemprop/chemprop/blob/b7385523aac79516b871f83feba5aeb4bc23c4da/tests/integration/test_regression_mol.py#L19) built in might have caught this issue?

**Comment by shihchengli** [comment]
We already have test cases for MVE and evidential methods (see [here](https://github.com/chemprop/chemprop/blob/b7385523aac79516b871f83feba5aeb4bc23c4da/tests/cli/test_cli_regression_rxn.py#L42) as an example). To capture the issue you found, we would need to test regression tasks without target normalization. AFAIK, we don't have a flag to disable target scaling, since the target should always be scaled to ensure stable model training ([we also scale targets in the test you refer to](https://github.com/chemprop/chemprop/blob/main/tests/integration/test_regression_mol.py#L37)).

**Comment by KnathanM** [comment]
@shihchengli  We scale the targets in the test he refers to, but we don't use the scaler from that to make an unscale transform for the FFN. 
In conftest we have both a normal mpnn and a dirichlet mpnn defined, both without an unscale transform in the FFN.
https://github.com/chemprop/chemprop/blob/f0ee198379a24f2de50335cbbf2c69bbb981d3d9/tests/integration/conftest.py#L11-L24

We use the `classification_mpnn_dirichlet` in `integration/test_classification_mol.py`
https://github.com/chemprop/chemprop/blob/f0ee198379a24f2de50335cbbf2c69bbb981d3d9/tests/integration/test_classification_mol.py#L100-L106

We could do something similar and have an MVE mpnn defined in `conftest.py` and use it in `integration/test_regression_mol.py`. Then we would test if MVE works without the unscale transform and the CLI tests will still test if MVE works with an unscale transform.

File: tests/integration/test_regression_mol.py [hunk 1]
```diff
@@ -10,18 +10,18 @@
...
-        [
-            (nn.BondMessagePassing(), nn.MeanAggregation()),
-            (nn.AtomMessagePassing(), nn.SumAggregation()),
-            (nn.BondMessagePassing(), nn.NormAggregation()),
-        ],
-        indirect=True,
-    ),
-    pytest.mark.integration,
-]
+# pytestmark = [
```
Comment (KnathanM) [inline tests/integration/test_regression_mol.py:13]: I think we can remove these lines instead of commenting them out.

**Review by KnathanM — APPROVED** [review]
I'll merge after we remove the commented out lines. @shihchengli if you aren't able to do this but agree with removing them, just say so and I can push that change.

**Review by KnathanM — APPROVED** [review]

**Pull request merged**